modules/internal/ChapelRange.chpl

```diff
@@ -1232,12 +1265,11 @@ operator :(r: range(?), type t: range(?)) {
 
   if tmp.stridable {
     tmp._stride = r.stride: tmp._stride.type;
-    tmp._alignment = r.alignment: tmp.intIdxType;
+    tmp._alignment = chpl__idxToInt(r.alignment).safeCast(tmp.intIdxType);;
```

Comment:
We need to normalize the alignment here.
```chpl
tmp._alignment = chpl__mod(chpl__idxToInt(r.alignment), tmp._stride): tmp._alignment.type;
```

Fyi, the return type of `chpl__mod` matches the type of its first argument. This computation follows what is done in the general range.init() ~line 281.

Ugh this normalization will break for castEnumRange4.chpl:
```chpl
enum color {red=42, orange=33};
var r1 = color.red..color.orange;
```

My initial reaction is that r1.alignment should be `0` internally and `red` if user facing. How about `r3 = .. align orange` ? Generally, alignment should correspond to how a range is iterated over, i.e., using the enum constants' ordinal numbers, not their integer values.

Reply:
> tmp._alignment = chpl__mod(chpl__idxToInt(r.alignment), tmp._stride): tmp._alignment.type;

I'm having trouble thinking of when this normalization would be necessary...  The stride doesn't seem like it could change due to the cast unless...  Oh, the old stride required more bits of precision than the new cast does?  OK, I guess I buy that and will try to create a test that exercises it (maybe something like `var r = (1..511 by -512): range(int(8), stridable=true;`?)

> My initial reaction is that r1.alignment should be 0 internally and red if user facing.

I agree with that...  Do you think it's not currently? (or wouldn't with the change above?)  Looks like it is [as of 1.30](https://ato.pxeger.com/run?1=m70sOSOxIDVnwYKlpSVpuhY3G1LzSnMVkvNz8osUqotSU3QU0otSU_N0FJJySlNrrblKi1MhstZcZYlFCkUKtgpAVXp6IGmFpEoFI2uu8qLMktScPI0ivcSczPS83NS8Ek1rLrBKBbAIxEgcCiHugDoH5iwA) and on main as well...?

Reply:
> OK, I guess I buy that and will try to create a test that exercises it (maybe something like var r = (1..511 by -512): range(int(8), stridable=true;?)

No, I failed because the stride downcast currently uses a safeCast so fails for cases like these.  So I can't come up with a case in which the alignment would be incorrect since the stride would be preserved (?).

Reply:
Yeah I missed a few details here. Thank you for clarifying the situation. Here is what I see now:

* My concern stemmed from observing _alignment=42 in a range(enum) cast. With this PR, this should no longer be the case.

* I do not see safeCast() on the stride conversion. Maybe it should be there? Or we can defer it to #19269.

* If the conversion of alignment currently on the branch, line 1270, i.e., `tmp._alignment = chpl__idxToInt(r.alignment).safeCast(tmp.intIdxType)`, simply transfers the value of `_alignment` from the old range to the new range, I would prefer to write it as `if r._alignment==none then 0 else r._alignment:tmp.intIdxType`. So much clearer! Note double `;` at the end of line 1270.

* If it is not a straight transfer... I need to think more.

* If it is a straight transfer, I do not see the need for saveCast() for alignment. Either way, see above.

Another case to consider is casting from a bigger enum to a smaller enum. Or some other situation where the old alignment may be greater than the highest ordinal of the new enum. This situation is actually independent of casting. Here are the cases that come to my mind:

```chpl
enum big { a=42, b, c, d, e=56};
enum small {a=42, e=56};
var r1 = big.a..big.e by 10 align big.e; // OK
var r2 = r1: range(small, stridable=true); // _alignment=4 is meaningless
var r3 = (42..56 by 10 align 8) :range(small, stridable=true); // ditto
```

Not sure what is here to be done. Perhaps add a check + throw if this occurs? We can leave it as-is for the purposes of this PR. Also if the new range is over bools.

Reply:
> I do not see safeCast() on the stride conversion. Maybe it should be there?

Huh, that's true—I must've gotten mixed up with the alignment's safeCast?  Let me look at this a bit more.

Reply:
OK, looking at it more, I'm remembering that we have a range.safeCast() method as well, which suggests that perhaps the `:` operator should continue to be unsafe w.r.t. stride and alignment values; though that raised the question for me about "What does 'safe' really mean in terms of our casts?" and to open  #22436.  I'm taking a break to work on CHIUW for a bit and will continue to stew on this.  At present, my thought is that stride and alignment probably should not use a safecast, to be consistent with integers, and that the enum cases should continue to throw, since the `:` on int->enum casts throws itself.

Reply:
Or, we could just remove the enum range casts altogether as some have proposed in #22406...